1-models/articles-model.js

```diff
@@ -11,13 +12,27 @@ exports.selectArticles = () => {
             articles.article_img_url,
             CAST(COUNT(comments.comment_id) AS INT) comment_count
         FROM articles
-        LEFT JOIN comments ON articles.article_id = comments.article_id
-        GROUP BY articles.author, articles.title, articles.article_id
-        ORDER BY articles.created_at DESC;`;
+        LEFT JOIN comments ON articles.article_id = comments.article_id 
+        `;
 
-  return db.query(queryString).then(({ rows }) => {
-    return rows;
-  });
+  if (topic) {
+    queryString += ` WHERE articles.topic = $1 GROUP BY articles.author, articles.title, articles.article_id
+    ORDER BY articles.created_at DESC;`;
+
+    return db.query(queryString, [topic]).then(({ rows }) => {
+      if (rows.length === 0) {
+        return Promise.reject({ status: 400, message: "Invalid Topic Query"})
+      }
+      return rows;
+    });
+  } else {
+    queryString += ` GROUP BY articles.author, articles.title, articles.article_id
+  ORDER BY articles.created_at DESC;`;
+
+    return db.query(queryString).then(({ rows }) => {
+      return rows;
+    });
+  }
```

Comment:
Really nice construction of your comment string here!

A small refactor you could consider - you need the `ORDER BY` section either way, so you could just only add the there topic bit inside your if statement, and then lose the else block.

Not something that you need to do, just something for you to consider for the future